xchange-bithumb/src/main/java/org/knowm/xchange/bithumb/service/BithumbTradeService.java

```diff
@@ -89,16 +86,20 @@ public String placeLimitOrder(LimitOrder limitOrder) throws IOException {
   @Override
   public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
 
-    if (orderParams instanceof CancelOrderByPairAndIdParams) {
-      try {
-        final CancelOrderByPairAndIdParams params = (CancelOrderByPairAndIdParams) orderParams;
-        return cancelBithumbOrder(params.getOrderId(), params.getCurrencyPair());
-      } catch (BithumbException e) {
-        throw BithumbErrorAdapter.adapt(e);
+    try {
```

Comment:
ok, if we are going in this direction:

```
CancelOrderByPairAndIdParams extends CancelOrderByIdParams, CancelOrderByCurrencyPair
```

then this method can be implemented in a much simpler way

```
  @Override
  public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
    try {
      if (!(orderParams instanceof CancelOrderByIdParams && orderParams instanceof CancelOrderByCurrencyPair)) {
                throw new NotYetImplementedForExchangeException(
                    "Only CancelOrderByPairAndIdParams || (CancelOrderByIdParams && CancelOrderByCurrencyPair) supported");
      }
      String orderId = ((CancelOrderByIdParams) orderParams).getOrderId();
      CurrencyPair pair = ((CancelOrderByCurrencyPair) orderParams).getCurrencyPair();
      return cancelBithumbOrder(orderId, pair);
    } catch (BithumbException e) {
      throw BithumbErrorAdapter.adapt(e);
    }
  }
```

Reply:
I totally agree, i just didn't want to obfuscate it at that time